### ISSUE TITLE
Updated gravity.db via script not reflected by Pihole Docker.

### ISSUE THREAD

**Issue by an0wak** [pi-hole/docker-pi-hole#1312]
* Using a cron job and scripts to manage enabling and disabling blacklist items at specific times. 
* Modifying the gravity.db works, but the changes are not reflected until services are manually restarted. 

This is a: I am not sure? but only happens under the Docker Container. Full Pihole running under Debian seems to work? Could be a Docker "thingy" or Pihole.

## Details

Understand PiHole was never intended for parental control but by far the easiest way to prevent access to social media sites, during school days.

I have a cron job run a script twice a week. Once on Monday morning and then on Friday afternoon. This script would enable or disable the gravity database blacklist table based on the comment with the word "script".

![image](https://user-images.githubusercontent.com/74270857/216712301-8e2de28a-3a15-4151-b4a7-a99433ab83db.png)

**enabledisable_domainlist.sh**
```
#!/bin/bash

dayofweek="$(date +%w)"
# Monday
if [ ${dayofweek} = "1" ]; then
 enabled=1
# Friday
elif [ ${dayofweek} = "5" ]; then
 enabled=0
else
 exit
fi

gravitydb="/etc/pihole/gravity.db"

# Enable/Disable blacklists with script in comment.
pihole-FTL sqlite3 ${gravitydb} "update 'domainlist' SET enabled = '${enabled}' WHERE comment like '%Script%';"

pihole restartdns reload-lists
```

Using docker-compose, I inject the initialisation script into /etc/cont-init.d to create the Cron jobs.

**Initialisation script**
_deploy-cron-job.sh_
```
#!/bin/bash
chmod +x /opt/script/enabledisable_domainlist.sh
crontab /opt/script/cron.txt
```

**Cron task.**
_cron.txt_
```
00 6 * * Mon /opt/script/enabledisable_domainlist.sh
30 16 * * Fri /opt/script/enabledisable_domainlist.sh
```

All the above works. Scripts are being triggered correctly and modify the DB as seen in the logs.

```
root@pihole:/var/log# cat cron.log
Fri Jan 20 16:30:01 AEDT 2023: 0
Fri Jan 20 16:30:01 AEDT 2023: /etc/pihole/gravity.db
Fri Jan 20 16:30:01 AEDT 2023
id  type  domain               enabled  date_added  date_modified  comment
--  ----  -------------------  -------  ----------  -------------  ---------------------
1   3     (.|^)youtube.*$      0        1661230986  1674192601      Script Block YouTube
2   3     (.|^)googlevideo.*$  0        1661230986  1674192601      Script Block YouTube
3   3     .*youtu.?be.*        0        1661230986  1674192601      Script Block YouTube
4   3     (.|^)tiktok.*$       0        1661230986  1674192601      Script Block TikTok
5   3     (.|^)tiktok.com$     0        1661230986  1674192601      Script Block TikTok
6   3     (.|^)twitch.tv$      0        1661230986  1674192601      Script Block Twitch
DNS Restarted
Mon Jan 23 06:00:01 AEDT 2023: 1
Mon Jan 23 06:00:01 AEDT 2023: /etc/pihole/gravity.db
Mon Jan 23 06:00:01 AEDT 2023
id  type  domain               enabled  date_added  date_modified  comment
--  ----  -------------------  -------  ----------  -------------  ---------------------
1   3     (.|^)youtube.*$      1        1661230986  1674414001      Script Block YouTube
2   3     (.|^)googlevideo.*$  1        1661230986  1674414001      Script Block YouTube
3   3     .*youtu.?be.*        1        1661230986  1674414001      Script Block YouTube
4   3     (.|^)tiktok.*$       1        1661230986  1674414001      Script Block TikTok
5   3     (.|^)tiktok.com$     1        1661230986  1674414001      Script Block TikTok
6   3     (.|^)twitch.tv$      1        1661230986  1674414001      Script Block Twitch
DNS Restarted
Fri Jan 27 16:30:01 AEDT 2023: 0
Fri Jan 27 16:30:01 AEDT 2023: /etc/pihole/gravity.db
Fri Jan 27 16:30:01 AEDT 2023
id  type  domain               enabled  date_added  date_modified  comment
--  ----  -------------------  -------  ----------  -------------  ---------------------
1   3     (.|^)youtube.*$      0        1661230986  1674797401      Script Block YouTube
2   3     (.|^)googlevideo.*$  0        1661230986  1674797401      Script Block YouTube
3   3     .*youtu.?be.*        0        1661230986  1674797401      Script Block YouTube
4   3     (.|^)tiktok.*$       0        1661230986  1674797401      Script Block TikTok
5   3     (.|^)tiktok.com$     0        1661230986  1674797401      Script Block TikTok
6   3     (.|^)twitch.tv$      0        1661230986  1674797401      Script Block Twitch
DNS Restarted
Mon Jan 30 06:00:01 AEDT 2023: 1
Mon Jan 30 06:00:01 AEDT 2023: /etc/pihole/gravity.db
Mon Jan 30 06:00:01 AEDT 2023
id  type  domain               enabled  date_added  date_modified  comment
--  ----  -------------------  -------  ----------  -------------  ---------------------
1   3     (.|^)youtube.*$      1        1661230986  1675018801      Script Block YouTube
2   3     (.|^)googlevideo.*$  1        1661230986  1675018801      Script Block YouTube
3   3     .*youtu.?be.*        1        1661230986  1675018801      Script Block YouTube
4   3     (.|^)tiktok.*$       1        1661230986  1675018801      Script Block TikTok
5   3     (.|^)tiktok.com$     1        1661230986  1675018801      Script Block TikTok
6   3     (.|^)twitch.tv$      1        1661230986  1675018801      Script Block Twitch
DNS Restarted
Fri Feb  3 16:30:01 AEDT 2023: 0
Fri Feb  3 16:30:01 AEDT 2023: /etc/pihole/gravity.db
Fri Feb  3 16:30:01 AEDT 2023
id  type  domain               enabled  date_added  date_modified  comment
--  ----  -------------------  -------  ----------  -------------  ---------------------
1   3     (.|^)youtube.*$      0        1661230986  1675402201      Script Block YouTube
2   3     (.|^)googlevideo.*$  0        1661230986  1675402201      Script Block YouTube
3   3     .*youtu.?be.*        0        1661230986  1675402201      Script Block YouTube
4   3     (.|^)tiktok.*$       0        1661230986  1675402201      Script Block TikTok
5   3     (.|^)tiktok.com$     0        1661230986  1675402201      Script Block TikTok
6   3     (.|^)twitch.tv$      0        1661230986  1675402201      Script Block Twitch
DNS Restarted
```

The issue I am having is that PiHole is ignoring the changes. It is only until I manually restart the DNS Service (under WebGUI or manually run *pihole restartdns reload-lists* till it works??.

Note: Pihole running directly under Debian and running these scripts works and reflects properly. Is there something under docker that causes this?

## Related Issues
- [ x] I have searched this repository/Pi-hole forums for existing issues and pull requests that look similar
- I have posted under Pihole community forum without success.

## How to reproduce the issue 

1. Environment data
  * Operating System: Debian 11.6 
  * Hardware: KVM VM under Debian 
  * Kernel Architecture: amd64 
  * Docker Install Info and version: 
    - Software source:  Docker version 20.10.22, build 3a2c30b 
    - Supplimentary Software: docker-compose version 1.29.2, build 5becea4c 
  * Hardware architecture: x86 

2. docker-compose.yml contents, docker run shell command, or paste a screenshot of any UI based configuration of containers here

```
version: "3.8"

services:
  dnscrypt:
    container_name: dnscrypt-proxy
    image: klutchell/dnscrypt-proxy:latest
    hostname: dnscrypt
    networks:
      pihole_net:
        ipv4_address: 172.16.255.2
    expose:
      - "5300/udp"
      - "5300/tcp"
    environment:
      TZ: ${TZ}
    volumes:
      - ./etc-dnscrypt-proxy:/config:rw
    restart: unless-stopped

  pihole:
    container_name: pihole
    image: pihole/pihole:latest
    dns:
      - 127.0.0.1
    hostname: ${HOSTNAME}
    domainname: ${DOMAIN_NAME}
    networks:
      pihole_net:
        ipv4_address: 172.16.255.3
    ports:
      - "53:53/tcp"
      - "53:53/udp"
      - "80:80/tcp"
      - "443:443/tcp"
    environment:
      TZ: ${TZ}
      WEBPASSWORD: ${WEBPASSWORD}
      REV_SERVER: ${REV_SERVER:-false}
      REV_SERVER_DOMAIN: ${REV_SERVER_DOMAIN}
      REV_SERVER_CIDR: ${REV_SERVER_CIDR}
      REV_SERVER_TARGET: ${REV_SERVER_TARGET}
      PIHOLE_DNS_: '172.16.255.2#5300'
      DNSSEC: 'true'
      DNSMASQ_LISTENING: single
    volumes:
      - ./etc-pihole:/etc/pihole:rw
      - ./etc-dnsmasq.d:/etc/dnsmasq.d:rw
      - ./opt-script/:/opt/script:rw
      - ./etc-cont-init/deploy-cron-job.sh:/etc/cont-init.d/99-deploy-cron-job.sh'
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    restart: unless-stopped
    depends_on:
      - dnscrypt

networks:
  pihole_net:
    driver: bridge
    ipam:
      config:
        - subnet: 172.16.255.0/24
```

3. any additional info to help reproduce
Don't think so, but it's doing my head in.

## These common fixes didn't work for my issue

- [ x] I have tried removing/destroying my container, and re-creating a new container
- [ x] I have tried fresh volume data by backing up and moving/removing the old volume data
- [ ] I have tried running the stock `docker run` example(s) in the readme (removing any customizations I added)
- [ ] I have tried a newer or older version of Docker Pi-hole (depending what version the issue started in for me)
- [ ] I have tried running without my volume data mounts to eliminate volumes as the cause

If the above debugging / fixes revealed any new information note it here.
Add any other debugging steps you've taken or theories on root cause that may help.

**Comment by PromoFaux** [comment]
Sorry, this passed me by somehow...

`pihole restartdns reload-lists`

Can you try changing that line to `service pihole-FTL restart`? The docker image uses S6 as the init system, and the `service` command we use is really just a wrapper around some of the s6 commands. It may just be that there is nothing that handles the `reload-lists` part.

**Comment by an0wak** [comment]
Hi PromoFaux,

Thank you for the suggestion, I will give that a go, but I have since reverted back to the Debian full installation and I seem to have the same issue. Changes made directly to the database, reflect in GUI, but PiHole ignores the changes. They seem to only work if done via the GUI? or database reloaded manually - very weird!

I've tried ```pihole``` stop and ```pihole start```. Had a suggestion to try using ```piHole restartGravity```

Let you know how I go.
